docs/client-sdk-states-and-lifecycle.mdx

```diff
@@ -0,0 +1,414 @@
+
+
+#### The "Enrolled" State
+
+In this state, the client has previously observed the experiment in **Enrolling** state and has been bucketed into a branch of the experiment. It persists the selected branch so that it can consistently reference it even if the experiment config changes in future.
+
+Transitions:
+
+
+
+*   To **Disqualified **state:
+    *   When the user explicitly opts out of participating in this experiment.
+    *   When the user explicitly opts out of participating in experiments in general.
+    *   When application code makes an explicit API call to disable this experiment.
+    *   When the experiment transitions to **Enrolling** or **Running** state, and its targeting expression no longer matches the client.
+        *   (N.B. this includes the experiment transitioning **Enrolling** -> **Enrolling** with an update to the experiment config, which can re-trigger bucketing evaluation)
+    *    When encountering some unrecoverable error in the experiment config.
+*   To **WasEnrolled **state:
+    *   When the experiment transitions to **Cancelled** or **Completed**
+        *   (i.e. When a valid response is received from Remote Settings and the experiment is no longer present in the Remote Settings collection).
+*   To **Discarded **state, when the user entirely disables telemetry in their client.
+
+
+#### The "Disqualified" State
+
+In this state, the client had previously enrolled in the experiment, but some change has occurred that means we can no longer show them to experiment treatment. The client will continue to tag its telemetry pings with enrollment data, but will not activate the treatment.
+
+Transitions:
+
+
+
+*   To **WasEnrolled **state:
+    *   When the experiment transitions to **Cancelled** or **Completed**
+        *   (i.e. When a valid response is received from Remote Settings and the experiment is no longer present in the Remote Settings collection).
+*   To **Discarded **state, when the user entirely disables telemetry in their client.
+
+Note that once a client has been disqualified from an experiment, it is not possible for it to be resume showing the experiment treatment.
+
+
+#### The "WasEnrolled" State
+
+In this state, the client remembers that it was previously enrolled in an experiment and may report diagnostic telemetry about this fact, but the experiment has ended and its treatments are no longer activated.
+
+WasEnrolled experiments are shown in about:studies:
+
+![WasEnrolled shown in about:studies](/img/client-sdk-states-and-lifecycle/WasEnrolled-in-about-studies.png
+"WasEnrolled shown in about:studies")
+
+
+It's also useful for analysis to have a period of observation (say 30 days?) after an experiment ends in which we still tag telemetry pings with the experiment branch info.
+
+Transitions:
+
+
+
+*   To **Discarded **state after 31 days have passed since it entered the **WasEnrolled** state, according to the client's local clock.
```

Comment:
Comment from the original doc:

> (@rfk) It may be more reliable to base this on a timestamp from the server

Reply:
>  if we can guarantee that the endDate is reliable and suitable for this

I don't think endDate is ever populated for anything published to RS so you'd need some behaviour changes.

Reply:
https://github.com/mozilla/experimenter/issues/5165 filed to track this further, marking this comment as resolved.